### DISCUSSION TITLE
Run migrations for custom schema defined in config by default

### DISCUSSION THREAD

**Discussion by unzico** [mikro-orm/mikro-orm#4982] [General]
### Describe the bug

If you use a custom schema and define it in the config that is passed to the MikroORM client, you would expect the migrations to be performed for the defined custom schema. Especially, when the migration is done programmatically. However, that's not the case. Instead, a migration that creates a table will create that table in the `public` schema by default.

**Example:**
```typescript
import { Migration } from "@mikro-orm/migrations";

export class Migration20231113142010 extends Migration {
  async up(): Promise<void> {
    const k = this.getKnex();

    await k.schema
      // The next line is required to create the table in a schema called 'custom'.
      // But that shouldn't be necessary if we define the schema in the config
      // of the MikroORM instance that programmatically runs the migrations.
      .withSchema("custom")
      .createTable("test", (t) => {
        t.increments("id").unsigned().primary();
      });
  }

  async down(): Promise<void> {
    const k = this.getKnex();

    await k.schema.dropTableIfExists("test");
  }
}

```

### Reproduction

https://github.com/unzico/mikroorm-schema-migration

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

"@mikro-orm/core": "5.9.4", "@mikro-orm/migrations": "5.9.4", "@mikro-orm/postgresql": "5.9.4", "@mikro-orm/reflection": "5.9.4"

### Node.js version

v20.5.1

### Operating system

macOS 14.1.1 (23B81)

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
You need to specify the schema yourself, migrations are not dynamic, they need to be self-contained. If you create a migration from the CLI, it will respect the schema from config - it's about that part, not about execution. Since you create the migration by hand, it is your responsibility.

**Reply by unzico** [reply]
So be it 👍
